src/ucs/memory/rcache_int.h

```diff
@@ -112,7 +137,76 @@ void ucs_mem_region_destroy_internal(ucs_rcache_t *rcache,
                                      ucs_rcache_region_t *region);
 
 
-#define UCS_RCACHE_STAT_MIN_POW2 \
-    ucs_roundup_pow2(ucs_global_opts.rcache_stat_min)
+static void __ucs_rcache_region_log(const char *file, int line, const char *function,
+                                    ucs_log_level_t level, ucs_rcache_t *rcache,
+                                    ucs_rcache_region_t *region, const char *fmt,
+                                    ...) UCS_F_PRINTF(7, 8);
+
+
+static inline int ucs_rcache_region_test(ucs_rcache_region_t *region, int prot)
+{
+    return (region->flags & UCS_RCACHE_REGION_FLAG_REGISTERED) &&
+           ucs_test_all_flags(region->prot, prot);
+}
+
+
+/* LRU spinlock must be held */
+static inline void
+ucs_rcache_region_lru_add(ucs_rcache_t *rcache, ucs_rcache_region_t *region)
+{
+    if (region->lru_flags & UCS_RCACHE_LRU_FLAG_IN_LRU) {
+        return;
+    }
+
+    ucs_rcache_region_trace(rcache, region, "lru add");
+    ucs_list_add_tail(&rcache->lru.list, &region->lru_list);
+    region->lru_flags |= UCS_RCACHE_LRU_FLAG_IN_LRU;
+}
+
+
+/* LRU spinlock must be held */
+static inline void
+ucs_rcache_region_lru_remove(ucs_rcache_t *rcache, ucs_rcache_region_t *region)
+{
+    if (!(region->lru_flags & UCS_RCACHE_LRU_FLAG_IN_LRU)) {
+        return;
+    }
+
+    ucs_rcache_region_trace(rcache, region, "lru remove");
+    ucs_list_del(&region->lru_list);
+    region->lru_flags &= ~UCS_RCACHE_LRU_FLAG_IN_LRU;
+}
+
+
+static void __ucs_rcache_region_log(const char *file, int line, const char *function,
+                                    ucs_log_level_t level, ucs_rcache_t *rcache,
+                                    ucs_rcache_region_t *region, const char *fmt,
+                                    ...)
```

Comment:
IMO ucs_rcache_region_test/lru_add/lru_remove should be in rcache.inl, since these are inline functions (also, consider making them UCS_F_ALWAYS_INLINE). 
Also, rename __ucs_rcache_region_log to ucs_rcache_region_log (remove underscore) and make it non-inline function declared in rcache_int.h and defined in rcache.c (it's slowpath, so does not need to be in h or inl file)